cmd/tegola/cmd/tile_name_format_test.go

```diff
@@ -0,0 +1,101 @@
+package cmd
+
+import (
+	"testing"
+	"reflect"
+	"fmt"
+)
+
+// TODO(ear7h): use internal package, currently in maths/internal
+func errOk(expected, got error) bool {
+	if expected == nil && got == nil {
+		return true
+	}
+
+	if expected != nil && got != nil {
+		return expected.Error() == got.Error()
+	}
+
+	return false
+}
+
+func TestNewFormat(t *testing.T) {
+	testcases := map[string]struct {
+		formatStr string
+		exepected Format
+		err       error
+	}{
+		"1": {
+			formatStr: "/zxy",
+			exepected: Format{1, 2, 0, "/"},
+			err:       nil,
+		},
+		"2": {
+			formatStr: " xyz",
+			exepected: Format{0, 1, 2, " "},
+			err:       nil,
+		},
+		"3 invalid formatStr": {
+			formatStr: "//zxy",
+			err:       fmt.Errorf("invalid formatStr //zxy"),
+		},
+		"4 invalid formatStr": {
+			formatStr: "1zxy",
+			err:       fmt.Errorf("invalid formatStr 1zxy"),
+		},
+	}
```

Comment:
Add the following test failure test cases:
"/1xy"
",z45"
"zzxy"
"xyyz
"$xxx"
"$yyy"
"$zzz"
"$xyx"
"$xyy"
"$xzz"
"$xzx"
"$yxx"
"$yxy"
"$xzz"
"$zzz"
"$$$$"
",100"